tests/Backend/Features/RealReadOnlyBranches/SnowflakeRealBranchesTest.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\Test\Backend\Features\RealReadOnlyBranches;
+
+use Keboola\StorageApi\ClientException;
+use Keboola\StorageApi\DevBranches;
+use Keboola\StorageApi\Workspaces;
+use Keboola\Test\Backend\Workspaces\ParallelWorkspacesTestCase;
+use Keboola\Test\Backend\Workspaces\Backend\WorkspaceBackendFactory;
+
+class SnowflakeRealBranchesTest extends ParallelWorkspacesTestCase
+{
+    private DevBranches $branches;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->branches = new DevBranches($this->_client);
+        $this->cleanupTestBranches($this->_client);
+    }
+
+    /**
+     * This test verifies visibility of buckets from workspaces across default and dev branches
+     * when project has storage-branches feature set only.
+     * Scenario:
+     *  1. Create bucket in default branch.
+     *  2. Create dev branch and create bucket inside it.
+     *  3. Assert backendPath for both buckets.
+     *  4. Create workspace in default branch – ensure dev bucket NOT visible.
+     *  5. Create workspace in dev branch – ensure BOTH buckets visible.
+     *  6. Drop dev branch – ensure its bucket and workspace are deleted (no longer accessible).
```

Comment:
chybi mi tam jeden case - vyrabis a operujes nad devbranchi X. Ale co jina branch Y, ktera by nemela videt bucket z branche X? V bode 4 to testujes jen default - dev. Ale tam se krizi projektova RO role a dev RO role. Myslim, ze potrebujeme case dev-dev